### PR TITLE
Conenctor builder: Record processing page

### LINKED ISSUE #24948: Builder documentation: Record selection page

### PR DESCRIPTION
**Pull request by flash1293**
## What

Fixes https://github.com/airbytehq/airbyte/issues/24948

### CODE REVIEW

File: docs/connector-development/connector-builder-ui/record-processing.mdx [hunk 1]
```diff
@@ -0,0 +1,339 @@
+import Diff from './assets/record-processing-schema-diff.png';
+
+# Record processing
+
+Builder connectors always make HTTP requests, receive the responses and emit records. Besides making the right requests, it's important to properly hand over the records to the system:
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/record-processing.mdx:5]: Is "Builder connector" a term we use elsewhere? I don't hate the name, but we should standardize what we call these connectors so avoid confusion

File: docs/connector-development/connector-builder-ui/record-processing.mdx [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+Very often, the response body contains an array of records along with some suplementary information (for example meta data for pagination).
+
+For example the ["Most popular" NY Times API](https://developer.nytimes.com/docs/most-popular-product/1/overview) returns the following response body:
+<pre>
+{`{
+    "status": "OK",
+    "copyright": "Copyright (c) 2023 The New York Times Company.  All Rights Reserved.",
+    "num_results": 20,
+    `}<b>{`"results": [`}</b>{`
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/record-processing.mdx:22]: does this get rendered as expected in the docs page?

File: docs/connector-development/connector-builder-ui/record-processing.mdx [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        "CAD": 1.739516
+    `}<b>{`}`}</b>{`
+}`}
+</pre>
+
+In this case, a record selector of `rates` will yield a single record which contains all the exchange rates in a single object.
+
+## Transformations
+
+It is recommended to not change records during the extraction process the connector is performing, but instead load them into the downstream warehouse unchanged and perform necessery transformations there in order to stay flexible in what data is required. However there are some reasons that require the modifying the fields of records before they are sent to the warehouse:
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/record-processing.mdx:106]: Agreed

File: docs/connector-development/connector-builder-ui/record-processing.mdx [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+  {
+    "code": "DE",
+    "type": "country"
+  },
+]
+```
+
+In this case, the "Primary key" can be set to "`code`, `type`" to allow automatic downstream deduplication of records based on the value of these two fields.
+
+### Declared schema
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/record-processing.mdx:254]: We encountered another edge case during the discourse build a connector session.

If a field is not set on any of the records read, the type will show up as "null", which is almost certainly wrong. Developers should make sure there's at least one valid type for all fields in the schema (and the UI should probably have a warning).

Updating the UI is out of scope here, but we should document this gotcha

**Review by girarda — APPROVED** [review]

File: docs/connector-development/connector-builder-ui/record-processing.mdx (hunk 4 above)
Comment (flash1293) [reply docs/connector-development/connector-builder-ui/record-processing.mdx:254]: Good point, added a paragraph for this

File: docs/connector-development/connector-builder-ui/record-processing.mdx (hunk 1 above)
Comment (flash1293) [reply docs/connector-development/connector-builder-ui/record-processing.mdx]: Not really, changed it to "Connectors built with the connector builder"

**Review by clnoll — APPROVED** [review]
LGTM!

**Pull request merged**